lib/shared/image_viewer.dart

```diff
@@ -370,14 +359,24 @@ class _ImageViewerState extends State<ImageViewer> with TickerProviderStateMixin
                                         _imageViewer.currentState?.clearSnackBars();
                                         _imageViewer.currentState?.showSnackBar(snackBar);
                                       });
+                                    } finally {
+                                      setState(() => isDownloadingMedia = false);
                                     }
                                   },
-                            icon: Icon(
-                              Icons.share_rounded,
-                              semanticLabel: "Comments",
-                              color: Colors.white.withOpacity(0.90),
-                              shadows: const <Shadow>[Shadow(color: Colors.black, blurRadius: 50.0)],
-                            ),
+                            icon: isDownloadingMedia
+                                ? SizedBox(
+                                    height: 20,
+                                    width: 20,
+                                    child: CircularProgressIndicator(
+                                      color: Colors.white.withOpacity(0.90),
+                                    ),
+                                  )
+                                : Icon(
+                                    Icons.share_rounded,
```

Comment:
Could you change this to "Share" instead of "Comments"? I think that was something that snuck in from a previous PR!